pkg/server/token_minter.go

```diff
@@ -72,11 +75,12 @@ type oidcClaims struct {
 
 // NewRouter creates a new HTTP server implementation that will exchange
 // a GitHub OIDC token for a GitHub application token with eleveated privlidges.
-func NewRouter(ctx context.Context, githubAppInstallation *githubauth.AppInstallation, configStore ConfigReader, jwtParseOptions []jwt.ParseOption) (*TokenMintServer, error) {
+func NewRouter(ctx context.Context, githubApp *githubauth.App, configStore ConfigReader, jwtParseOptions []jwt.ParseOption) (*TokenMintServer, error) {
 	return &TokenMintServer{
-		githubAppInstallation: githubAppInstallation,
-		configStore:           configStore,
-		jwtParseOptions:       jwtParseOptions,
+		githubApp:         githubApp,
+		configStore:       configStore,
+		jwtParseOptions:   jwtParseOptions,
+		githubAppInstalls: cache.New[*githubauth.AppInstallation](1 * time.Hour),
```

Comment:
You actually don't need to cache these; they are cached automatically by the library.

Reply:
Oh, great. I'll remove that then.

Reply:
I'm curious - the cache in pkg https://github.com/abcxyz/pkg/blob/main/githubauth/app.go#L268-L282 seems to only cache "installations" but it seems we want to cache access tokens (so we can reuse them before timeout)?

Reply:
> I'm curious - the cache in pkg https://github.com/abcxyz/pkg/blob/main/githubauth/app.go#L268-L282 seems to only cache "installations" but it seems we want to cache access tokens (so we can reuse them before timeout)?

Correct - the githubapp package caches the lookup of an installation ID from inputs. It does not cache any generated tokens.

Reply:
Caching the access token would be problematic since each access token might have a different set of permissions it is requesting. We'd need to devise a cache key that encapsulated the set of permissions or something. Its better to request a fresh access token each time rather than risk using the wrong token and giving more permissions than required.